### ISSUE TITLE
Feat invalidate transport

### ISSUE THREAD

**Issue by nithril** [nithril/smtp-connection-pool#19]
From https://github.com/nithril/smtp-connection-pool/pull/18
This PR adds the possibility to invalidate ClosableSmtpConnection instead of returning them in the pool.

My use case is:
- I do not want to testOnBorrow or testOnReturn since this impacts performance negatively
- If there is no exception when sending emails, return the connection to the pool, it appears to still be good
- If there is an exception, do not return the connection to the pool, it might be bad

**Comment by nithril** [comment]
@evigeant, could you please check if the commit https://github.com/nithril/smtp-connection-pool/commit/750d6b77c071ce828125ff9bb807ad3dbb0fbf7c is ok for you?

**Comment by evigeant** [comment]
Yes, it all makes sense, cleaned up the code nicely. I did not know, you could have this kind of code:

```
void test() throws SpecificException {
  try {
    // Code that throws SpecificException
  } catch(Exception e) {
    throw e;
  }
}
```
With older versions of the JDK, it would complain about the throws clause I believe, but I learned today that it works well... Thanks.

**Comment by nithril** [comment]
Released in 1.4.0
